caikit_nlp/modules/text_generation/text_generation.py

```diff
@@ -0,0 +1,261 @@
+        error.type_check("<FPT48137045E>", str, allow_none=True, eos_token=eos_token)
+        error.type_check("<FPT53511308E>", str, allow_none=True, pad_token=pad_token)
+        self.base_model = base_model
+        self.base_model_name = base_model_name
+
+        # Set _model_loaded as False by default. This will only get set to True if
+        # we enable the tgis_backend and we are able to fetch the client successfully.
+        self._model_loaded = False
+        # Configure the internal client
+        # NOTE: This is made optional for the cases where we do not need to execute `.run` function
+        # for example, bootstrapping a model to caikit format and saving.
+        if tgis_backend:
+            self._client = tgis_backend.get_client(base_model_name)
+            # mark that the model is loaded so that we can unload it later
+            self._model_loaded = True
+
+        self._bos_token = bos_token
+        self._sep_token = sep_token
+        self._eos_token = eos_token
+        self._pad_token = pad_token
+
+    def __del__(self):
+        # nothing to unload if we didn't finish loading
+        if self._model_loaded:
+            self.get_backend().unload_model(self._model_path)
+
+    @classmethod
+    def bootstrap(cls, base_model_path: str, load_backend: BackendBase = None):
+        """Function to bootstrap a pre-trained transformers model and
+        get a caikit text-generation 'model'.
+
+        Args:
+            base_model_path: str
+                Path to transformers model
+                NOTE: Model path needs to contain tokenizer as well
+            load_backend: BackendBase
+                Backend object to be used to run inference with.
+                NOTE: this is required for inferencing. It is
+                made optional just in provide support for model conversion use-case
+        Returns:
+            caikit_nlp.blocks.text_generation.TextGeneration
+                Object of TextGeneration class (model)
+        """
+        # pylint: disable=duplicate-code
+        model_config = AutoConfig.from_pretrained(base_model_path)
+
+        resource_type = None
+        for resource in cls.supported_resources:
+            if model_config.model_type in resource.SUPPORTED_MODEL_TYPES:
+                resource_type = resource
+                break
+
+        if not resource_type:
+            error(
+                "<FPT61784225E>",
+                "{} model type is not supported currently!".format(
+                    model_config.model_type
+                ),
+            )
+        log.debug("Bootstrapping base resource [%s]", base_model_path)
+        base_model = resource_type.bootstrap(
+            base_model_path, tokenizer_name=base_model_path
+        )
+        bos_token = base_model._tokenizer.bos_token
+        sep_token = base_model._tokenizer.sep_token
+        eos_token = base_model._tokenizer.eos_token or None
```

Comment:
Why does only EOS have a fallback here? Shouldn't any fallback behavior be handled in bootstrap of the base model?

Reply:
this is just to assign eos_token to something that gets ignored and the decision of "ending the sequence" is then left to the model.